### PR TITLE
Add branch listing

### PR DESCRIPTION
**Pull request by mfb2**
This commit adds the ability to fetch a list of branches from Bitbucket
via pagination.

### CODE REVIEW

File: repository.go [hunk 1]
```diff
@@ -107,6 +127,30 @@ func (r *Repository) GetFileBlob(ro *RepositoryBlobOptions) (*RepositoryBlob, er
...
+	if rbo.Sort != "" {
+		params.Add("sort", rbo.Sort)
+	}
+
+	if rbo.PageNum > 0 {
+		params.Add("page", strconv.Itoa(rbo.PageNum))
+	}
+
+	urlStr := r.c.requestUrl("/repositories/%s/%s/refs/branches?%s", rbo.Owner, rbo.RepoSlug, params.Encode())
+	response, err := r.c.executeRaw("GET", urlStr, "")
```
Comment (mfb2) [inline repository.go]: The existing call to `execute` presumed pagination, but for some reason came back malformed.  I opted to manually map the data on the response instead.

File: repository.go [hunk 2]
```diff
@@ -274,6 +318,53 @@ func decodeRepositoryFiles(repoResponse interface{}) ([]RepositoryFile, error) {
...
+		err = mapstructure.Decode(branchEntry, &branch)
+		if err == nil {
+			branches = append(branches, branch)
+		}
+	}
+
+	page, ok := branchResponseMap["page"].(float64)
+	if !ok {
+		page = 0
+	}
```
Comment (mfb2) [inline repository.go]: I'm curious as to your thoughts on this section.  This set of conditionals that evaluate the metadata (`page`, `pagelen`, `size`) in the response from Bitbucket should always be valued; however, if something is missing, the current behavior will default it to a value of `0`.  I'm not certain how consumers would treat or view this response; any insights on a possibly better idea?

**Comment by mfb2** [comment]
Hello!  I tested this and it works well for me, though I'd sleep much better if someone else could validate/confirm.  Please let me know if you have any feedback or questions, I'm happy to help out!

File: repository.go (hunk 2 above)
Comment (ktrysmt) [reply repository.go]: It is no problem, I think. At the struct definition, `0` is correct If you use int types as `Page int`.
(https://github.com/ktrysmt/go-bitbucket/pull/66/files#diff-ab2c7209d94fb144c8d55246ad821816R47)

So gopher would be care to use to struct and zero types. @mfb2

**Comment by mfb2** [comment]
Thank you for approving!  I am happy to contribute, and there's a chance I may submit a few more improvements in the near future.  Please let me know if I can provide any other assistance – I look forward to the merge!  😄

**Pull request merged**